Cleaning up ipsets failed with --cleanup-config option

Cleaning up ipsets failed with `--cleanup-config` option due to
    invalid ipset handler.
Same issue in network_policy_controller.go and network_routes_controller.go

    {"log":"panic: runtime error: invalid memory address or nil pointer dereference\n","stream":"stderr"}
    {"log":"[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xf8429c]\n","stream":"stderr""}
    {"log":"github.com/cloudnativelabs/kube-router/pkg/utils.(*IPSet).DestroyAllWithin(0x0, 0xc, 0x19e758c)\n","stream":"stderr"}
    {"log":"\u0009/root/go/src/github.com/cloudnativelabs/kube-router/pkg/utils/ipset.go:278 +0x2c\n","stream":"stderr"}
    {"log":"github.com/cloudnativelabs/kube-router/pkg/controllers/netpol.(*NetworkPolicyController).Cleanup(0xc42068bb60)\n","stream":"stderr"}
    {"log":"\u0009/root/go/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:1417 +0x8cf\n","stream":"stderr"}

i also found same problem

$ docker run --rm --privileged --net=host cloudnativelabs/kube-router --version 
Running /usr/local/bin/kube-router version v0.2.0-beta.6, built on 2018-05-17T16:39:27+0000, go1.8.7

```
$ docker run --privileged --net=host cloudnativelabs/kube-router --cleanup-configI0604 10:30:52.851061       1 network_policy_controller.go:1343] Cleaning up iptables configuration permanently done by kube-router
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xfa5c4c]

goroutine 1 [running]:
github.com/cloudnativelabs/kube-router/pkg/utils.(*IPSet).DestroyAllWithin(0x0, 0xc, 0x1a441ba)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/utils/ipset.go:278 +0x2c
github.com/cloudnativelabs/kube-router/pkg/controllers/netpol.(*NetworkPolicyController).Cleanup(0xc4205e7b68)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/controllers/netpol/network_policy_controller.go:1417 +0x517
github.com/cloudnativelabs/kube-router/pkg/cmd.CleanupConfigAndExit()
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/pkg/cmd/kube-router.go:66 +0x5f
main.Main(0x26cf620, 0x0)
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/cmd/kube-router/kube-router.go:51 +0x2d6
main.main()
	/home/travis/gopath/src/github.com/cloudnativelabs/kube-router/cmd/kube-router/kube-router.go:17 +0x26
```

--- a/docs/user-guide.md
+++ b/docs/user-guide.md
@@ -95,9 +95,10 @@ kube-router --master=http://192.168.1.99:8080/ --run-firewall=true --run-service
 
 ## cleanup configuration
 
-You can clean up all the configurations done (to ipvs, iptables, ip routes) by kube-router on the node by running
+Please delete kube-router daemonset and then clean up all the configurations done (to ipvs, iptables, ipset, ip routes etc) by kube-router on the node by running below command.
+
 ```
- kube-router --cleanup-config
+docker run --privileged --net=host cloudnativelabs/kube-router --cleanup-config
 ```
 
 ## trying kube-router as alternative to kube-proxy

--- a/pkg/controllers/netpol/network_policy_controller.go
+++ b/pkg/controllers/netpol/network_policy_controller.go
@@ -1414,7 +1414,15 @@ func (npc *NetworkPolicyController) Cleanup() {
 	}
 
 	// delete all ipsets
-	err = npc.ipSetHandler.DestroyAllWithin()
+	ipset, err := utils.NewIPSet()
+	if err != nil {
+		glog.Errorf("Failed to clean up ipsets: " + err.Error())
+	}
+	err = ipset.Save()
+	if err != nil {
+		glog.Errorf("Failed to clean up ipsets: " + err.Error())
+	}
+	err = ipset.DestroyAllWithin()
 	if err != nil {
 		glog.Errorf("Failed to clean up ipsets: " + err.Error())
 	}

--- a/pkg/controllers/routing/network_routes_controller.go
+++ b/pkg/controllers/routing/network_routes_controller.go
@@ -446,7 +446,16 @@ func (nrc *NetworkRoutingController) Cleanup() {
 		glog.Warningf("Error deleting Pod egress iptable rule: %s", err.Error())
 	}
 
-	err = nrc.ipSetHandler.DestroyAllWithin()
+	// delete all ipsets created by kube-router
+	ipset, err := utils.NewIPSet()
+	if err != nil {
+		glog.Errorf("Failed to clean up ipsets: " + err.Error())
+	}
+	err = ipset.Save()
+	if err != nil {
+		glog.Errorf("Failed to clean up ipsets: " + err.Error())
+	}
+	err = ipset.DestroyAllWithin()
 	if err != nil {
 		glog.Warningf("Error deleting ipset: %s", err.Error())
 	}

